cmd/samproxy/main.go

```diff
@@ -60,7 +60,11 @@ func main() {
 		os.Exit(0)
 	}
 
-	c, err := config.NewConfig(opts.ConfigFile, opts.RulesFile)
+	a := app.App{
+		Version: version,
+	}
+
+	c, err := config.NewConfig(opts.ConfigFile, opts.RulesFile, func(err error) { a.Logger.Error().WithField("error", err).Logf("error reloading config") })
```

Comment:
I wonder if it's worth checking that the logger is not nil here? I think the chances are pretty slim that the config reloads in the time before the logger gets assigned, but better to be safe than sorry?